grr/server/grr_response_server/gui/ui/components/code_editor/code_editor.ts

```diff
@@ -0,0 +1,55 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ViewChild,
+    ElementRef,
+    AfterViewInit,
+    ViewEncapsulation,
+    Output,
+    EventEmitter,
+    Input,
+} from '@angular/core';
+
+import * as CodeMirror from 'codemirror';
+// importing sql-hint is needed for SQL syntax highlighting
+import 'codemirror/addon/hint/sql-hint.js';
+// importing show-hint is needed for the autocomplete pop-up
+import 'codemirror/addon/hint/show-hint.js';
+
+
+/** Displays a code editor. */
+@Component({
+  selector: 'code-editor',
+  templateUrl: './code_editor.ng.html',
+  styleUrls: ['./code_editor.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class CodeEditor implements AfterViewInit {
+  @ViewChild('editorTarget')
+  private readonly editorTarget!: ElementRef;
+
+  @Input() initialValue = '';
```

Comment:
Yeap I was planning to extend it with this functionality in the next PR containing the query helper, but we can also do it here.

It looks like the `propName` and `propNameChange` convention is useful when I want to bind a **variable** in the parent component to some property in the child component. However, here I need the stream of changes (not just the latest value in a variable), because I am calling `this.form.patchValue({ query: latestValue });` (and I didn't find a way to use two-way binding with form controls, something like `[(value)]="form.query"`?).

Can we make it so the parent component sets the value of the child code editor through a Subject like `[value]="setEditorValue$ | async"`, which is initialized in the following way: `readonly setEditorValue$ = new BehaviorSubject(this.defaultQueryDisplayed);`. 
On the other hand, the child code editor exposes a `valueChange` event emitter, to which the parent component listens to and updates the `form.query` value: `(valueChange)="updateQuery($event)"`. In the child component we can update the values as follows:

```
private latestValue = '';
  @Input()
  set value(newText: string) {
    this.latestValue = newText;

    this.valueChange.emit(this.latestValue);
  }
  get value(): string {
    return this.latestValue;
  }

  @Output() valueChange = new EventEmitter<string>();
```